test/e2e/app-dir/rsc-basic.test.ts

```diff
@@ -64,6 +64,45 @@ describe('app dir - rsc basics', () => {
     return
   }
 
+  it('should correctly render page returning null', async () => {
+    const homeHTML = await renderViaHTTP(next.url, '/return-null/page')
+    const $ = cheerio.load(homeHTML)
+    expect($('#return-null-layout').html()).toBeEmpty()
+  })
+
+  it('should correctly render component returning null', async () => {
+    const homeHTML = await renderViaHTTP(next.url, '/return-null/component')
+    const $ = cheerio.load(homeHTML)
+    expect($('#return-null-layout').html()).toBeEmpty()
+  })
+
+  it('should correctly render layout returning null', async () => {
+    const homeHTML = await renderViaHTTP(next.url, '/return-null/layout')
+    const $ = cheerio.load(homeHTML)
+    expect($('#return-null-layout').html()).toBeEmpty()
+  })
+
+  it('should correctly render page returning undefined', async () => {
+    const homeHTML = await renderViaHTTP(next.url, '/return-undefined/page')
+    const $ = cheerio.load(homeHTML)
+    expect($('#return-undefined-layout').html()).toBeEmpty()
+  })
+
+  it('should correctly render component returning undefined', async () => {
+    const homeHTML = await renderViaHTTP(
+      next.url,
+      '/return-undefined/component'
+    )
+    const $ = cheerio.load(homeHTML)
+    expect($('#return-null-undefined').html()).toBeNull()
```

Comment:
Is this behavior expected?
Rendering component returning `undefined` behaves differently than rendering `undefined` on a page. That's not true for `null`.

Reply:
![image](https://user-images.githubusercontent.com/12820484/205932543-27b2a753-400f-4440-aeec-c521d1fc95e1.png)
:facepalm: